contracts/contracts/token/OUSD.sol

```diff
@@ -584,20 +633,94 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
             ? MAX_SUPPLY
             : _newTotalSupply;
 
-        _rebasingCreditsPerToken = _rebasingCredits.divPrecisely(
-            _totalSupply.sub(nonRebasingSupply)
-        );
+        _rebasingCreditsPerToken =
+            (_rebasingCredits * 1e18) /
+            (_totalSupply - nonRebasingSupply);
 
         require(_rebasingCreditsPerToken > 0, "Invalid change in supply");
 
-        _totalSupply = _rebasingCredits
-            .divPrecisely(_rebasingCreditsPerToken)
-            .add(nonRebasingSupply);
+        _totalSupply =
+            ((_rebasingCredits * 1e18) / _rebasingCreditsPerToken) +
+            nonRebasingSupply;
 
         emit TotalSupplyUpdatedHighres(
             _totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to)
+        external
+        onlyGovernor
+        nonReentrant
+    {
+        require(from != to, "Cannot delegate to self");
+        require(
+            yieldFrom[to] == address(0) &&
+                yieldTo[to] == address(0) &&
+                yieldFrom[from] == address(0) &&
+                yieldTo[from] == address(0),
+            "Blocked by existing yield delegation"
+        );
+        RebaseOptions stateFrom = rebaseState[from];
+        RebaseOptions stateTo = rebaseState[to];
+        require(
+            _isNonRebasingAccount(from) &&
+                (stateFrom == RebaseOptions.NotSet ||
+                    stateFrom == RebaseOptions.StdNonRebasing),
+            "Must delegate from a non-rebasing account"
```

Comment:
@DanielVF  in most cases the account delegating yield will be a contract account that will by default have the rebaseState set to `StdNonRebasing`.  Some might not though, and if this call is done for an EOA account it feels annoying one must opt out of rebasing while waiting for the governance transaction that enables yield delegation to pass.  Also since the yield delegation shall only be called by the governance it feels like such decision should overwrite any previous states and preferences. 

For that reason I would change this to an if statement and force a rebaseOptOut of the `from` account if such is not the state already.

Reply:
Yes, I think we could change the code to automatically convert NonSet rebasing accounts to non-rebasing.

Reply:
done here: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/a5269b87b88d52193d5af7bd29530556bd3e4d82